jihyeon/app.js

```diff
@@ -107,6 +78,100 @@ app.post('/users/posts', async function (req, res) {
     res.status(201).json({message: 'postCreated'})
 });
 
+app.post('/users/:userId/likes/:postId', async function (req, res) {
+    const { userId, postId } = req.params
+
+    await dataSource.query(`
+        INSERT INTO likes (
+            user_id,
+            post_id
+        ) VALUES (
+            ?,
+            ?
+        )
+    `,[ userId, postId ])
+
+    res.status(201).json({message: 'likeCreated'})
+});
+
+app.get('/posts/viewall', async function (req, res){
+    await dataSource.query(
+        `SELECT
+            u.id AS user_id,
+            p.id AS post_id,
+            p.post_image,
+            p.post_content
+        FROM posts p 
+        INNER JOIN users u ON p.user_id = u.id`
+                , function (err, rows) {
+        res.status(200).json(rows);
+          });
+});
+
+app.get('/users/:userId/posts', async function (req, res){
+    const { userId } = req.params;
+    
+    const [ showPosts ] = await dataSource.query(
+        `SELECT
+            u.id as user_id,
+            JSON_ARRAYAGG(
+                JSON_OBJECT(
+                    "post_id", p.id,
+                    "post_image", p.post_image,
+                    "post_content", p.post_content
+              )
+            ) as posts
+        FROM users u 
+        JOIN posts p ON p.user_id = u.id
+        WHERE u.id = ?
+        GROUP BY u.id`, [ userId ]
+    );
+
+    return res.status(200).json({data : showPosts})
+});
+
+app.put('/users/:userId/posts/:postId/update', async function (req, res){
+    const { userId, postId } = req.params
+    const { postImage, postContent } = req.body
+
+    await dataSource.query(
+        `UPDATE users u INNER JOIN posts p
+         ON u.id = p.user_id
+            SET
+                post_image = ?,
+                post_content = ?
+            WHERE u.id = ? AND p.id = ?
+        `,
+        [ postImage, postContent, userId, postId ]
+    );
+    
+     const [result] = await dataSource.query(
+        `SELECT
+            u.id as userId,
+            u.name as userName,
+            p.id as postingId,
+            p.post_image as postingImage,
+            p.post_content as postingContent
+        FROM users AS u 
+        JOIN posts p ON p.user_id = u.id
+        WHERE u.id = ? AND p.id = ?
+        `, [ userId, postId ]
+    );
+    
+    res.status(200).json({ data : result })
+});
+
+app.delete('/users/:userId/posts/delete/:postId', async function (req, res){
```

Comment:
```suggestion
app.delete('/users/:userId/posts/:postId', async function (req, res){
```
delete라는 동작은 http method를 통해 이미 충분히 설명되므로 지워주세요!